### DISCUSSION TITLE
Optional: the getStimParams function can return parameters that are not the best.

### DISCUSSION THREAD

**Discussion by kurokida** [kurokida/jsQuestPlus#4] [Ideas]
Hi, @tpronk 

I'm still not sure how important this is, but I updated to receive an argument for the getStimParams function to return parameters that are not the best.

https://github.com/kurokida/jsQuestPlus/commit/1c4e2f9ad584c9a375a778c7294820fd5b3d04a7#diff-d1141079a0d00258fe71f84dc63e820796cf66e886b73b46895ae340a4ad4b2d

USAGE

```javascript
const most_valuable_stim = jsqp.getStimParams()
const most_valuable_stim2 = jsqp.getStimParams(1) // This is the same as most_valuable_stim
const third_valuable_stim = jsqp.getStimParams(3) 
```

The most valuable stimulus parameters are when the entropy is at its minimum.
However, the observer may feel burdened by the response to such stimuli because the observer cannot easily determine the response.
Perhaps for that reason, another QUEST+ library also has a parameter called randomFromBestN.
https://github.com/BrainardLab/mQUESTPlus/blob/2482d0481732f16a32e0c42a6f130116e7d9ee63/questplus/qpRun.m#L59

**Comment by tpronk** [comment]
If the main purpose is to deliver a stimulus that is randomly picked from a set of stimuli, maybe have `getStimParams` return an array? Something like this perhaps?
```
// Return a single stimulus
const most_valuable_stim = jsqp.getStimParams()
// Return an array with one stimulus; same stimulus as above
const most_valuable_stim = jsqp.getStimParams(1)
// Return an array with the three best stimuli
const most_valuable_stim2 = jsqp.getStimParams(3)
```

**Reply by kurokida** [reply]
Thank you for your reply.

The getStimParams needs to return a single stimulus-parameter combination, not an array.

If a random value is needed, we can write as the following

```javascript
const tmp = Math.round(Math.random()*10)
const stim_params = jsQuestPlus.getStimParams(tmp)
```

This is a bit off topic, but we should write jsQuestPlus.getStimParams, not jsqp.getStimParams. I made a mistake in my first post.
